### PR TITLE
数据源是oracle时 默认命名空间返回时null 校验不通过

### PR DESCRIPTION
**Pull request by Aeizzz**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

数据源是oracle时 默认命名空间返回时null 校验不通过

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: config/src/main/java/com/alibaba/nacos/config/server/service/HistoryService.java [hunk 1]
```diff
@@ -102,10 +103,19 @@ public List<ConfigInfoWrapper> getConfigListByNamespace(String namespaceId) {
...
+            throw new AccessException("Please check dataId.");
+        }
+        if(!Objects.equals(configHistoryInfo.getGroup(), group)){
+            throw new AccessException("Please check group.");
+        }
+        if((!StringUtils.isBlank(configHistoryInfo.getTenant()) && !Objects
+                .equals(configHistoryInfo.getTenant(), namespaceId))){
+            throw new AccessException("Please check namespaceId.");
+        }
+        if(StringUtils.isNotEmpty(configHistoryInfo.getTenant()) && StringUtils.isNotEmpty(namespaceId)){
```
Comment (KomachiSion) [inline config/src/main/java/com/alibaba/nacos/config/server/service/HistoryService.java:117]: isNotEmpty and !isBlank I think duplicate.

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: config/src/main/java/com/alibaba/nacos/config/server/service/HistoryService.java (hunk 1 above)
Comment (Aeizzz) [reply config/src/main/java/com/alibaba/nacos/config/server/service/HistoryService.java:117]: 第一个是判断 Tenant 不是空值时候 两个进行相等的判断，第二种是 Tenant 是空但是 namespaceId 不为空的判断。
isNotEmpty 和 !isBlank 是相同的作用，主要是 Objects.equals 两个值比较如果第一个值 是null 第二个值 “”空串的情况下是有问题的

Comment (KomachiSion) [reply config/src/main/java/com/alibaba/nacos/config/server/service/HistoryService.java:117]: 那第二个逻辑写的应该有问题。
tenant是空，namespaceId不为空应该是`StringUtils.isEmpty(configHistoryInfo.getTenant()) && StringUtils.isNotEmpty(namespaceId))`

Comment (KomachiSion) [reply config/src/main/java/com/alibaba/nacos/config/server/service/HistoryService.java:117]: 而且不需要写这么复杂， 直接比较namespaceId和tenant是否相等就可以了。一方空一方不空肯定就不一致啊。没必要判断两种。

Comment (Aeizzz) [reply config/src/main/java/com/alibaba/nacos/config/server/service/HistoryService.java:117]: 这里判断一个空一个不为空，是 mysql 数据库下 tenant 可以是空串，namespaceId也是空串，直接比较是相等的， oracle 数据库下 tenant 数据库中没有空串，查询出来是 null ，和 namespaceId 空串比较是不一致的，导致历史配置查看的时候查询不出来。

**Comment by Rigavinil201** [comment]
>

**Pull request closed without merging**